3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp

```diff
@@ -53,6 +55,14 @@ class pipeline_token_stream final
   virtual bool next() override;
   virtual bool reset(const string_ref& data) override;
 
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief calls visitor on pipeline members in respective order. Visiting is 
+  /// interrupted on first visitor returning false.
+  /// @param visitor visitor to call
+  /// @return true if all visits returned true, false otherwise
+  //////////////////////////////////////////////////////////////////////////////
+  bool visit_members(const std::function<bool(const irs::analysis::analyzer::ptr)>& visitor) const;
```

Comment:
maybe use template-based visitor?

Reply:
That may create unnecessary complexity  for shared builds. But if we drop shared builds than that will be more flexible of course.